generator/ServiceClientGeneratorLib/Generators/Marshallers/RestXmlRequestMarshaller.tt

```diff
@@ -115,6 +115,14 @@ namespace <#=this.Config.Namespace #>.Model.Internal.MarshallTransformations
 #>
 				xmlWriter.WriteStartElement("<#=operation.RequestStructure.MarshallName#>");
 				xmlWriter.WriteAttributeString("xmlns", "<#=operation.XmlNamespacePrefix#>", null, "<#=operation.XmlNamespace#>");
+<#+
+			}
+			var requestStructureAttributes = operation.RequestStructure.Members.Where(x => x.IsXmlAttribute);
+			foreach(var member in requestStructureAttributes)
+			{
+#>
+				if(<#=variableName + ".IsSet" + member.PropertyName#>())
```

Comment:
Nit but I'd add a space between the `,` so that the generated code looks like:
```
xmlWriter.WriteAttributeString("test", "", StringUtils.FromString(publicRequest.Payload.Attr))
```

Instead of:
```
xmlWriter.WriteAttributeString("test","",StringUtils.FromString(publicRequest.Payload.Attr))
```